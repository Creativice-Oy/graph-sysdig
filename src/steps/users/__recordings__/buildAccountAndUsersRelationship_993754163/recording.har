{
  "log": {
    "_recordingName": "buildAccountAndUsersRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "0d29cdc1ea9a8e8f7ab793f1e65b94a7",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "us2.app.sysdig.com"
            }
          ],
          "headersSize": 331,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://us2.app.sysdig.com/api/user/me"
        },
        "response": {
          "bodySize": 2644,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json;charset=utf-8",
            "size": 2644,
            "text": "[\"1f8b0800000000000000c456db72db3610fd950efb58c906295ef5544552533771ec316de7a193f140e492420d022a00fa528fffbd0b40d4a549db342fcd83232e16c0d9b3670ff912f41a54307d090ca84ecf443d97a2668649a183a9513d8c02c33af8430a08a6c10f844c090946c18655a65770a33846d7c66ca6a7a78f8f8f27ada20fd4507552c9eed4ff3c85ba0ee3b408735224590510257596251368629ac7349ce07155af8dec4095600c13adb680f4b3ae596b7f81a02b0e75306d28d70868fb5c96cb5d68d557f76070dfaf9f5e111da7c26eece8d3ac0561e361320aa4584047453dc4c8c8665c03edf0611cfa4a391360cf79091a25bb602a7ace71450ebf34ed369883c042e2ffbd8efe3d37fd86dcaf4a9efca7ec3c8d77e99fb07a308a55fa9075108d5415ec88d56bf9b87cda80426a84c1d86e85b38e19b7e7b7ee29982604e9d4861a5d7b6a4601dd6ce66ba8eeb78b1b0407660d3d6ecac971e412d4a5921568bd4daea410506d7598fbdc76d6b60a5a6aa373d90b831745fb6bbebc6aef38f7757e04d6ae314c4e88a567f313e3a087541f5ac003ab8e826ba94d09ea01014ae594e4d1198a4251472bc9969477fd0a940003fa0ab4ec91ce3d6bf7bbb54b59ebad880ec3bfc8d53ebcbb67014233f31c4c231bee00d43d3565cf8c1d0b3ba6afa3a1754edd075c0c63eca70659eee694573da7b8f74cb4a07dd2d16cd9acb2527403b7d1f1feb75cae287f97eb19e758c17c00385488383454680dcb6168fdee4e0a66a45afe659429e7b2b248fc4c0ebc2393ba5fe94ab18d4557a2acacfd6017a91332b6953e5b41ea3361004dc7669dd5568c0cff06719126459aa598680c77f666ff4bc8ab7507d4021e72e94cc2ea03b3189a9e42bdcdb746b403e8aeb4a91f686721f8e2c67835283c48c318352f6aaaea319668d6fcd9815346d843820fd637b7a72c45bd707584e9242f8a9064688be4b8d25b64d2b50395db315b18d4d7f2365a38743b541245475b98698d1363795872d6326476df05100f4c496117b17e5b77b5abcdb3841a8bf2b81805c295e6bd82567608df014a2d28c2248fe3221c174d968fe32cc9c614423a0e434209cd1b42d21c8b8327e442506ef90f1a92e74d34a1639ae0c61892669c17151de3be3a4a62baca9ce1d758d85c0175fa453ed24951908c4cb202814ada9bf55e22561967026946c551e5ccfae55b705692732451aa595d2be734d876abff71afa313749113ffcab1efaec3743bdde8ca713cc1a0359a3928c31a56b9667a796bcd8f048f5528e8a4818f0a67740f5b3bbfd803f8c2cd4ea17263ef00b7674df5dd0a40dc316ca939ba4183b9a45a3f4a753052b57f7f537e25b97b95a1d31b7cc90d8f2feee96c6bd32989885fdf2afcdc4fea771708817a03f6ebd76b7009df676fde90371146edb7c3ee9c2c497c647b0eca8fb6baf9b1ed28e35b4e95b4120dae2ede2fefae97b3f3bbf3d987d9dbe5556061774c0ca57d06313c8258ba21fc3f1122a59c6a73b3a97712ceb230226984faf6f29ee1dbeb612ff0384de33c2693e49b668c59bf418437ae9e9fa9c6f908d2a6a9a39c24a4c0dbab2a8ea2103faf569388d2bac8e38ca409547113ad621a134aa3ac098bb84a421a610ee4d61d7bc17eefe1ece8743c79a0ebe42bc719dba1704aeda7d461cbe0f81df070606ef64bc17e08d8575cc35487595f76850849cd533713755fb94fbca05ccc31bb5c94c127df50f1f70dc5c3b5197463a041d3f7bddbc66eed1e0ce1041a703332b47f7e535e5f9cbbd6dbfc1247f04278e97dd6f07f00c00ee4f7fafa27000000ffff03008f6d00c9710b0000\"]"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "server",
              "value": "nginx/1.14.2"
            },
            {
              "name": "date",
              "value": "Wed, 24 Nov 2021 00:55:04 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 576,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-24T00:55:04.038Z",
        "time": 765,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 765
        }
      },
      {
        "_id": "4803564bbeac8b9c969b5288b919bccb",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "us2.app.sysdig.com"
            }
          ],
          "headersSize": 357,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://us2.app.sysdig.com/api/v2/users/light?limit=250&offset=0"
        },
        "response": {
          "bodySize": 606,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json;charset=utf-8",
            "size": 606,
            "text": "[\"1f8b0800000000000000a450cb6ec23010fc973d5b9113880d395101a73e9070db4b85909b6ca85bc7467e2055887faf93a24a48e550f5b28799d9d99d3942f4e83c542f47500d5439a594f1b2247048b0b206aa820c1a233b840a0ecac99d6f67bb4e2a9dd5b6030268e4abc6b41d5c4402fed307ecd656f7faf5ea6eb99d3f89c7d5fd729db4ad723e3c7c7b8980ad3409d4f2077beefdcf9040342b23b08e2e51391b719e1794153c1f116864c0b9c3349b8163a3e9b448ec84a50f820c318582da9a74b04b12027b679b58873e2b88c53c21622160732217c9c75793cbe88332d9076a8d6fe866ef71af023a6b306bf0f08f1a6e06e3cb1a6ecf57ae35c1c68c4df2829793df9ba09c963cff63131b02c106a987e4b66d3d06a8e8e90b0000ffff03006178942a24020000\"]"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "server",
              "value": "nginx/1.14.2"
            },
            {
              "name": "date",
              "value": "Wed, 24 Nov 2021 00:55:05 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 576,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-24T00:55:04.847Z",
        "time": 669,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 669
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
