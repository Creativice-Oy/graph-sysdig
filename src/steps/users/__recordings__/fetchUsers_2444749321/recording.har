{
  "log": {
    "_recordingName": "fetchUsers",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "4803564bbeac8b9c969b5288b919bccb",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json;charset=UTF-8"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "us2.app.sysdig.com"
            }
          ],
          "headersSize": 357,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://us2.app.sysdig.com/api/v2/users/light?limit=250&offset=0"
        },
        "response": {
          "bodySize": 606,
          "content": {
            "_isBinary": true,
            "mimeType": "application/json;charset=utf-8",
            "size": 606,
            "text": "[\"1f8b0800000000000000a450cb6ec23010fc973d5b9113880d395101a73e9070db4b85909b6ca85bc7467e2055887faf93a24a48e550f5b28799d9d99d3942f4e83c542f47500d5439a594f1b2247048b0b206aa820c1a233b840a0ecac99d6f67bb4e2a9dd5b6030268e4abc6b41d5c4402fed307ecd656f7faf5ea6eb99d3f89c7d5fd729db4ad723e3c7c7b8980ad3409d4f2077beefdcf9040342b23b08e2e51391b719e1794153c1f116864c0b9c3349b8163a3e9b448ec84a50f820c318582da9a74b04b12027b679b58873e2b88c53c21622160732217c9c75793cbe88332d9076a8d6fe866ef71af023a6b306bf0f08f1a6e06e3cb1a6ecf57ae35c1c68c4df2829793df9ba09c963cff63131b02c106a987e4b66d3d06a8e8e90b0000ffff03006178942a24020000\"]"
          },
          "cookies": [
            {
              "httpOnly": true,
              "name": "INGRESSCOOKIEAPI",
              "path": "/",
              "value": "[REDACTED]"
            }
          ],
          "headers": [
            {
              "name": "server",
              "value": "nginx/1.14.2"
            },
            {
              "name": "date",
              "value": "Wed, 24 Nov 2021 00:54:39 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json;charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "vary",
              "value": "Origin, Access-Control-Request-Method, Access-Control-Request-Headers, Accept-Encoding, User-Agent"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block"
            },
            {
              "name": "cache-control",
              "value": "no-cache, no-store, max-age=0, must-revalidate"
            },
            {
              "name": "pragma",
              "value": "no-cache"
            },
            {
              "name": "expires",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "DENY"
            },
            {
              "name": "content-encoding",
              "value": "gzip"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=15768000"
            },
            {
              "_fromType": "array",
              "name": "set-cookie",
              "value": "[REDACTED]"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 576,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-11-24T00:54:39.319Z",
        "time": 792,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 792
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
