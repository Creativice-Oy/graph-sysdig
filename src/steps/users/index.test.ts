import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { buildAccountAndUsersRelationship, fetchUsers } from '.';
import { integrationConfig } from '../../../test/config';
import { setupSysdigRecording } from '../../../test/recording';
import { fetchAccountDetails } from '../account';
import { Relationships } from '../constants';

describe('#fetchUsers', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSysdigRecording({
      directory: __dirname,
      name: 'fetchUsers',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchUsers(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_user'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'sysdig_user' },
          name: { type: 'string' },
          username: { type: 'string' },
          displayName: { type: 'string' },
          id: { type: 'string' },
          dateCreated: { type: 'number' },
          enabled: { type: 'boolean' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
          lastSeenOnSecure: { type: 'number' },
          products: { type: 'array', items: { type: 'string' } },
          status: { type: 'string' },
          systemRole: { type: 'string' },
          version: { type: 'number' },
        },
      },
    });
  });
});

describe('#buildAccountAndUsersRelationship', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSysdigRecording({
      directory: __dirname,
      name: 'buildAccountAndUsersRelationship',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should build users and account relationship', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchAccountDetails(context);
    await fetchUsers(context);
    await buildAccountAndUsersRelationship(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      collectedEntities: context.jobState.collectedEntities,
      encounteredTypes: context.jobState.encounteredTypes,
      collectedRelationships: context.jobState.collectedRelationships,
    }).toMatchSnapshot();

    const account = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_account'),
    );
    expect(account.length).toBe(1);
    expect(account).toMatchGraphObjectSchema({
      _class: ['Account'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'sysdig_account' },
          name: { type: 'string' },
          username: { type: 'string' },
          displayName: { type: 'string' },
          id: { type: 'string' },
        },
      },
    });

    const users = context.jobState.collectedEntities.filter((e) =>
      e._type.includes('sysdig_user'),
    );
    expect(users.length).toBeGreaterThan(0);
    expect(users).toMatchGraphObjectSchema({
      _class: ['User'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'sysdig_user' },
          name: { type: 'string' },
          username: { type: 'string' },
          displayName: { type: 'string' },
          id: { type: 'string' },
          dateCreated: { type: 'number' },
          enabled: { type: 'boolean' },
          firstName: { type: 'string' },
          lastName: { type: 'string' },
          lastSeenOnSecure: { type: 'number' },
          products: { type: 'array', items: { type: 'string' } },
          status: { type: 'string' },
          systemRole: { type: 'string' },
          version: { type: 'number' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.ACCOUNT_HAS_USER._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'sysdig_account_has_user',
          },
        },
      },
    });
  });
});
