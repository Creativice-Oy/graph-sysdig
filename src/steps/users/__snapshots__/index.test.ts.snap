// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#buildAccountAndUsersRelationship should build users and account relationship 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Account",
      ],
      "_key": "sysdig_account:10006755",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "accessKey": "91584491-9f78-4757-ae1a-110a0a8f0068",
            "additionalRoles": Array [],
            "agentInstallParams": Object {
              "accessKey": "91584491-9f78-4757-ae1a-110a0a8f0068",
              "checkCertificate": true,
              "collectorAddress": "ingest-us2.app.sysdig.com",
              "collectorPort": 6443,
              "sslEnabled": true,
            },
            "currentTeam": 10006021,
            "customer": Object {
              "accessKey": "91584491-9f78-4757-ae1a-110a0a8f0068",
              "dateCreated": 1636399070379,
              "externalId": "f088f23a-a591-4e5f-89ca-e1ad254ab713",
              "id": 1002849,
              "name": null,
            },
            "customerSettings": Object {
              "environment": Object {},
              "plan": Object {
                "allocatedAgentsCount": 15,
                "indirectCustomer": false,
                "maxAgents": 15,
                "maxTeams": -1,
                "metricsSettings": Object {
                  "enableGlobalK8sAllPodContainers": false,
                  "enablePromCalculatedIngestion": false,
                  "enablePromScrapeV2": true,
                  "enforce": false,
                  "enforceAgentAggregation": true,
                  "limits": Object {
                    "appCheck": 500,
                    "appCheckAggregationCount": 12,
                    "connections": 80,
                    "containerDensity": 200,
                    "containerServerPorts": 5,
                    "hostServerPorts": 10,
                    "jmx": 500,
                    "kubernetesJobs": 10000,
                    "kubernetesPods": 10000,
                    "limitKubernetesResources": false,
                    "meerkatSuited": true,
                    "progAggregationCount": 12,
                    "promMetricsWeight": 0,
                    "prometheus": 8000,
                    "prometheusPerProcess": 500,
                    "statsd": 1000,
                    "topDevicesCount": 10,
                    "topFilesCount": 10,
                  },
                  "showExperimentals": false,
                },
                "migratedToV2Direct": false,
                "monitorEnabled": false,
                "onDemandAgents": 0,
                "overageAssessmentEligible": false,
                "partner": "None",
                "paymentsIntegrationId": Object {
                  "id": "49659676",
                  "ttl": Object {
                    "ttl": 50,
                  },
                },
                "pricingPlan": "pro",
                "secureEnabled": true,
                "subscriptionState": "trialing",
                "subscriptionVersion": 2,
                "timelines": Array [
                  Object {
                    "from": null,
                    "sampling": 10000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 60000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 600000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 3600000000,
                    "to": null,
                  },
                  Object {
                    "from": null,
                    "sampling": 86400000000,
                    "to": null,
                  },
                ],
                "trialEndDate": 1638991071460,
                "trialPlanName": "secure-enterprise-standard-monthly",
              },
              "sysdig": Object {
                "buckets": Array [],
                "enabled": false,
                "enabledSSE": false,
              },
            },
            "dateActivated": 1636466484035,
            "dateCreated": 1636399220686,
            "enabled": true,
            "firstName": "Stefan",
            "id": 10006755,
            "intercomUserIdHash": "6ffd280509637cc4221ceeb32aad9847065ec4f2b4a40aa27f194c51a2b32e86",
            "lastName": "Virag",
            "lastSeenOnSecure": 1637712062713,
            "lastUpdated": 1637712062713,
            "name": "viragsf@gmail.com",
            "oauth": false,
            "pictureUrl": "http://www.gravatar.com/avatar/edd1469180957cee25d7753ef4a84a13",
            "products": Array [
              "SDC",
              "SDS",
            ],
            "properties": Object {
              "has_been_invited": true,
            },
            "remoteWriteParams": Object {
              "serverAddress": "us2.app.sysdig.com",
            },
            "resetPassword": false,
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "teamRoles": Array [
              Object {
                "admin": true,
                "role": "ROLE_TEAM_MANAGER",
                "teamId": 10006020,
                "teamName": "Monitor Operations",
                "teamTheme": "#7BB0B2",
                "userId": 10006755,
                "userName": "viragsf@gmail.com",
              },
              Object {
                "admin": true,
                "role": "ROLE_TEAM_MANAGER",
                "teamId": 10006021,
                "teamName": "Secure Operations",
                "teamTheme": "#7BB0B2",
                "userId": 10006755,
                "userName": "viragsf@gmail.com",
              },
            ],
            "termsAndConditions": true,
            "timezone": "+00:00",
            "uniqueIntercomUserId": "10006755.f088f23a-a591-4e5f-89ca-e1ad254ab713",
            "username": "viragsf@gmail.com",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_account",
      "active": undefined,
      "createdOn": undefined,
      "displayName": "Stefan Virag",
      "id": "10006755",
      "name": "viragsf@gmail.com",
      "status": "confirmed",
      "username": "viragsf@gmail.com",
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sysdig_user:10006755",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": 1636399220686,
            "enabled": true,
            "firstName": "Stefan",
            "id": 10006755,
            "lastName": "Virag",
            "lastSeenOnSecure": 1637712062713,
            "products": Array [
              "SDC",
              "SDS",
            ],
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "username": "viragsf@gmail.com",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_user",
      "active": undefined,
      "createdOn": undefined,
      "dateCreated": 1636399220686,
      "displayName": "Stefan Virag",
      "enabled": true,
      "firstName": "Stefan",
      "id": "10006755",
      "lastName": "Virag",
      "lastSeenOnSecure": 1637712062713,
      "name": "viragsf@gmail.com",
      "products": Array [
        "SDC",
        "SDS",
      ],
      "status": "confirmed",
      "systemRole": "ROLE_CUSTOMER",
      "username": "viragsf@gmail.com",
      "version": 2,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sysdig_user:10006754",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": 1636399070571,
            "enabled": true,
            "firstName": "Austin",
            "id": 10006754,
            "lastName": "Kelleher",
            "lastSeenOnSecure": 1636466812758,
            "products": Array [
              "SDC",
              "SDS",
            ],
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "username": "austin.kelleher@jupiterone.dev",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_user",
      "active": undefined,
      "createdOn": undefined,
      "dateCreated": 1636399070571,
      "displayName": "Austin Kelleher",
      "enabled": true,
      "firstName": "Austin",
      "id": "10006754",
      "lastName": "Kelleher",
      "lastSeenOnSecure": 1636466812758,
      "name": "austin.kelleher@jupiterone.dev",
      "products": Array [
        "SDC",
        "SDS",
      ],
      "status": "confirmed",
      "systemRole": "ROLE_CUSTOMER",
      "username": "austin.kelleher@jupiterone.dev",
      "version": 2,
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sysdig_account:10006755",
      "_key": "sysdig_account:10006755|has|sysdig_user:10006755",
      "_toEntityKey": "sysdig_user:10006755",
      "_type": "sysdig_account_has_user",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "sysdig_account:10006755",
      "_key": "sysdig_account:10006755|has|sysdig_user:10006754",
      "_toEntityKey": "sysdig_user:10006754",
      "_type": "sysdig_account_has_user",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "sysdig_account",
    "sysdig_user",
    "sysdig_account_has_user",
  ],
  "numCollectedEntities": 3,
}
`;

exports[`#fetchUsers should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sysdig_user:10006755",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": 1636399220686,
            "enabled": true,
            "firstName": "Stefan",
            "id": 10006755,
            "lastName": "Virag",
            "lastSeenOnSecure": 1637712062713,
            "products": Array [
              "SDC",
              "SDS",
            ],
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "username": "viragsf@gmail.com",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_user",
      "active": undefined,
      "createdOn": undefined,
      "dateCreated": 1636399220686,
      "displayName": "Stefan Virag",
      "enabled": true,
      "firstName": "Stefan",
      "id": "10006755",
      "lastName": "Virag",
      "lastSeenOnSecure": 1637712062713,
      "name": "viragsf@gmail.com",
      "products": Array [
        "SDC",
        "SDS",
      ],
      "status": "confirmed",
      "systemRole": "ROLE_CUSTOMER",
      "username": "viragsf@gmail.com",
      "version": 2,
    },
    Object {
      "_class": Array [
        "User",
      ],
      "_key": "sysdig_user:10006754",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "dateCreated": 1636399070571,
            "enabled": true,
            "firstName": "Austin",
            "id": 10006754,
            "lastName": "Kelleher",
            "lastSeenOnSecure": 1636466812758,
            "products": Array [
              "SDC",
              "SDS",
            ],
            "status": "confirmed",
            "systemRole": "ROLE_CUSTOMER",
            "username": "austin.kelleher@jupiterone.dev",
            "version": 2,
          },
        },
      ],
      "_type": "sysdig_user",
      "active": undefined,
      "createdOn": undefined,
      "dateCreated": 1636399070571,
      "displayName": "Austin Kelleher",
      "enabled": true,
      "firstName": "Austin",
      "id": "10006754",
      "lastName": "Kelleher",
      "lastSeenOnSecure": 1636466812758,
      "name": "austin.kelleher@jupiterone.dev",
      "products": Array [
        "SDC",
        "SDS",
      ],
      "status": "confirmed",
      "systemRole": "ROLE_CUSTOMER",
      "username": "austin.kelleher@jupiterone.dev",
      "version": 2,
    },
  ],
  "encounteredTypes": Array [
    "sysdig_user",
  ],
  "numCollectedEntities": 2,
}
`;
